frontend/src/network/GraphChangeButton.tsx

```diff
@@ -22,15 +22,37 @@ const GraphChangeButton = ({
 
   const buttonText = (): string => (isUpdating ? 'Updating' : isFollowing ? 'Unfollow' : 'Follow');
 
+  const getGraphChangeState = async (referenceId: string) => {
+    const operationStatus = await dsnpLink.graphOperationStatus(getContext(), { referenceId });
+    if (operationStatus.status !== 'pending') {
+      // Defined in App.tsx to refreshFollowing, triggers an api call to /graph/{msaId}/following
+      triggerGraphRefresh();
+      setIsUpdating(false);
+    } else {
+      setIsUpdating(true);
+    }
+    console.log('operationStatus', operationStatus);
+    return operationStatus;
+  };
+
   const changeGraphState = async () => {
-    setIsUpdating(true);
+    let refId: { referenceId?: string | undefined };
     if (isFollowing) {
-      await dsnpLink.graphUnfollow(getContext(), { msaId: connectionAccount.msaId });
+      refId = await dsnpLink.graphUnfollow(getContext(), { msaId: connectionAccount.msaId });
     } else {
-      await dsnpLink.graphFollow(getContext(), { msaId: connectionAccount.msaId });
+      refId = await dsnpLink.graphFollow(getContext(), { msaId: connectionAccount.msaId });
+    }
```

Comment:
code golf:
```suggestion
const refId = await (isFollowing ? dsnpLink.graphUnfollow(getContext(), { msaId:  connectionAccount.msaId }) : dsnpLink.graphFollow(getContext(), { msaId; connectionAccount.msaId }));
```

Reply:
☝🏻 note, if you apply the above suggestion you'll have to change a few lines below to remove `refId.`

Reply:
resolved this. while taking a second look at it, I made a couple other improvements